verified_boot_main.md

```diff
@@ -74,5 +74,46 @@
 
 ### ISOs
 
+ISO is format of disk file image containing exact duplicate of data from source
+disk. ISO disk images are common way to share software (including OS)
+installation media[^1]. This chapter will focus on various scenarios in which
+installation media could be tampered with, leading to compromised software
+being installed.
+
+`.iso` images cannot be modified by mounting them and modifying the files, yet
+there are ways to "modify" .iso image. One of the ways is repackaging, this
+means extracting image contents, performing modifications and repackaging
+contents into new disk image file. It is important to always check signatures,
+as described in [checksum verification chapter](#checksum-verification)[^2],
+to verify integrity of disk images. If package is skillfully repackaged, it's
+execution won't be prohibited by secure boot mechanism[^3]. Another way of
+compromising read-only attribute of `iso` images is flashing such image onto
+physical media.
+
+#### Verifying boot media with secure-boot
+
+Kicksecure would like to utilize secure boot to verify distributed installation media, yet there are limitations to this technology which do not allow it:
+* Booting `.iso` file directly, rather than flashing the contents onto physical
+media, is possible with boot-loaders like GRUB[^4]. Yet, such image cannot be
+cryptographically verified as a whole. GRUB verifies contents at
+component-level. This approach might still lead to executing repackaged `iso`.
+* Vast majority of x86 based hardware comes preloaded with Microsoft keys, this
+means that when secure boot is enabled, ony Microsoft signed binaries are
+allowed to run[^5]. Binaries can be signed only with a single key, this means
+that OS vendors cannot sign binaries themselves as they'd be prohibited from
+executing. Kicksecure came out with few ideas on how these limitations could be
+resolved:
+    * Maintain two disk images (one signed, one unsigned),
+    * Create disk images with two boot-loaders,
+    * Sell custom hardware with preloaded Kicksecure keys.
```

Comment:
Not true, multiple certificates are allowed, and each certificate may have a signature. It was that way since at least UEFI spec rev 2.0, which is first public revision. sbsigntools supports it since 2014: https://git.kernel.org/pub/scm/linux/kernel/git/jejb/sbsigntools.git/commit/src/image.c?id=f6115a8045275a0dc138f9088ba018441146e81d

Reply:
Glad, that I got corrected on this one. Important to point out. This might make some things easier in the future. (Signed with Microsoft compatible key + other, more secure key at the same time.)